keras_nlp/models/roberta_test.py

```diff
@@ -0,0 +1,104 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for Roberta model."""
+
+import os
+
+import tensorflow as tf
+from tensorflow import keras
+
+from keras_nlp.models import roberta
+
+
+class RobertaTest(tf.test.TestCase):
+    def test_valid_call_roberta(self):
+        model = roberta.Roberta(
+            vocabulary_size=50265,
+            num_layers=12,
+            num_heads=12,
+            hidden_dim=768,
+            intermediate_dim=3072,
+            max_sequence_length=12,
+            name="encoder",
+        )
+        input_data = {
+            "input_ids": tf.random.uniform(
+                shape=(1, 12), dtype=tf.int64, maxval=50265
+            ),
+            "input_mask": tf.constant(
+                [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0], shape=(1, 12)
+            ),
+        }
+        model(input_data)
+
+    def test_valid_call_classifier(self):
+        model = roberta.Roberta(
+            vocabulary_size=50265,
+            num_layers=12,
+            num_heads=12,
+            hidden_dim=768,
+            intermediate_dim=3072,
+            max_sequence_length=12,
+            name="encoder",
+        )
+        input_data = {
+            "input_ids": tf.random.uniform(
+                shape=(1, 12), dtype=tf.int64, maxval=50265
+            ),
+            "input_mask": tf.constant(
+                [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0], shape=(1, 12)
+            ),
+        }
+        classifier = roberta.RobertaClassifier(model, 4, 768, name="classifier")
+        classifier(input_data)
+
+    def test_valid_call_bert_base(self):
+        model = roberta.RobertaBase(name="encoder")
+        input_data = {
+            "input_ids": tf.random.uniform(
+                shape=(1, 512), dtype=tf.int64, maxval=model.vocabulary_size
+            ),
+            "input_mask": tf.constant([1] * 512, shape=(1, 512)),
+        }
+        model(input_data)
+
+    def test_saving_model(self):
+        model = roberta.Roberta(
+            vocabulary_size=50265,
+            num_layers=12,
+            num_heads=12,
+            hidden_dim=768,
+            intermediate_dim=3072,
+            max_sequence_length=12,
+            name="encoder",
+        )
+        input_data = {
+            "input_ids": tf.random.uniform(
+                shape=(1, 12), dtype=tf.int64, maxval=50265
+            ),
+            "input_mask": tf.constant(
+                [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0], shape=(1, 12)
+            ),
+        }
+        model_output = model(input_data)
+        save_path = os.path.join(self.get_temp_dir(), "model")
+        model.save(save_path)
+        restored_model = keras.models.load_model(save_path)
+
+        restored_output = restored_model(input_data)
+        self.assertAllClose(
+            model_output["sequence_output"],
+            restored_output["sequence_output"],
+            atol=1e-5,
```

Comment:
I assume you had to raise the tol to avoid flakiness! I had a hard time getting sequence output to work for Bert and switch to pooled (not an option here).